modules/oe_whitelabel_extra_project/oe_whitelabel_extra_project.module

```diff
@@ -144,5 +160,147 @@ function _oe_whitelabel_extra_project_preprocess_inpage_nav(array &$variables):
       'label' => $node->{$item['#field_name']}->getFieldDefinition()->getLabel(),
     ];
   }
+}
+
+/**
+ * Adds variables for the project status.
+ *
+ * @param array $variables
+ *   Variables from hook_preprocess_node().
+ */
+function _oe_whitelabel_extra_project_preprocess_status_and_progress(array &$variables): void {
+  /** @var \Drupal\node\NodeInterface $node */
+  $node = $variables['node'];
+  /** @var \Drupal\datetime_range\Plugin\Field\FieldType\DateRangeItem|null $date_range_item */
+  $date_range_item = $node->get('oe_project_dates')->first();
+  if ($date_range_item === NULL || !$date_range_item->value || !$date_range_item->end_value) {
```

Comment:
I don't mind the thoroughness, but just FYI the field has validation so either both are empty or both have value

Reply:
up 2 u

Reply:
The notorious question about defensive code in Drupal.
I don't trust these outside world expectations enough to just ignore them, but I also don't trust them enough to log an error.
So I end up with silent suppression of unexpected cases.

Reply:
Also depends if a website will change the config.
In that case we could log a warning or just ignore and not render the thing.